src/controllers/accomodation.controller.js

```diff
@@ -105,6 +105,89 @@ class Accommodation {
     const query = await accomodationServices.bookAccommodation(req.user.id, req.body.accommodationId, req.body.roomId, req.body.departureDate, req.body.checkoutDate);
     return response.successMessage(res, 'Booking was successfully processed', 201, query);
   }
+
+  /**
+   * This function helps user to rate accommodation which he booked.
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return accommodation response
+   */
+  static async rateAccomodation(req, res) {
+    try {
+      const { rate } = req.body;
+      const userId = req.user.id;
+      const { accomodationId } = req.params;
+
+      const rateData = { rate, userId, accomodationId };
+
+      const rateAccomodation = await accomodationServices.CreateAccomodationRate(rateData);
+      return response.successMessage(res, 'You rated this accomodation successfully', 201, rateAccomodation);
+    } catch (error) {
+      return response.errorMessage(res, error.message, 500);
+    }
+  }
+
+  /**
+   * This function helps user to update accommodation rate.
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return accommodation response
+   */
+  static async updateAccomodationRate(req, res) {
+    try {
+      const { rate } = req.body;
+      const userId = req.user.id;
+      const { accomodationId } = req.params;
+      await accomodationServices.updateAccomodationRate(accomodationId, userId, rate);
+      const updatedRate = await accomodationServices.getAccommodationRate(accomodationId, userId);
+      return response.successMessage(res, 'Rate was updated successfully', 201, updatedRate);
+    } catch (error) {
+      return response.errorMessage(res, error.message, 500);
+    }
+  }
+
+  /**
+   * This function helps user to get accommodation rate
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return accommodation response
+   */
+  static async getAccommodationRate(req, res) {
+    try {
+      const userId = req.user.id;
+      const { accomodationId } = req.params;
+      const accommodationRate = await accomodationServices.getAccommodationRate(accomodationId, userId);
+      return response.successMessage(res, 'Accommodation rate', 201, accommodationRate);
+    } catch (error) {
+      return response.errorMessage(res, error.message, 500);
+    }
+  }
+
+  /**
+   * This function helps user to rate accommodation which he booked.
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return accommodation response
+   */
+  static async getAverageRatings(req, res) {
```

Comment:
This function should only get the average ratings of a particular accommodation.

Reply:
Yeah I changed it now it is getting average only.